presto-hive/src/main/java/com/facebook/presto/hive/HiveConnectorFactory.java

```diff
@@ -135,6 +138,13 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
             HiveAnalyzeProperties hiveAnalyzeProperties = injector.getInstance(HiveAnalyzeProperties.class);
             ConnectorAccessControl accessControl = new PartitionsAwareAccessControl(injector.getInstance(ConnectorAccessControl.class));
             Set<Procedure> procedures = injector.getInstance(Key.get(new TypeLiteral<Set<Procedure>>() {}));
+            ConnectorPlanOptimizerProvider planOptimizerProvider = new ConnectorPlanOptimizerProvider() {
```

Comment:
Could you actually create a `ConnectorPlanOptimizerProvider` implementation that returns `ImmutableSet.of()`? It's better to have that concrete class instead of creating a lambda here. Inject that class and use `injector` to get the instance here like others do.